community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -234,7 +235,16 @@ public void init()
                     break;
                 case POPULATING:
                     // The database was shut down during population, or a crash has occurred, or some other sad thing.
-                    indexProxy = proxySetup.createRecoveringIndexProxy( descriptor, providerDescriptor, constraint );
+                    if ( constraint && indexRule.getOwningConstraint() == null )
```

Comment:
The autoboxing rule for `==` is that if one of the sides is a primitive, the boxed object will be unboxed (rather than the primitive being boxed). The bytecode of (this comparison in) `getOwningConstraint()` confirms this:

    ALOAD 0
    GETFIELD org/neo4j/kernel/impl/store/record/IndexRule.owningConstraint : Ljava/lang/Long;
    INVOKEVIRTUAL java/lang/Long.longValue ()J
    LDC -1
    LCMP
    IFNE L3